proposed/dotnet-bootstrapper/dotnet-bootstrapper-path.md

```diff
@@ -0,0 +1,130 @@
+
+2. On Windows, a registry key is set in `HKLM` (the registry hive for local user contexts). `bootstrapper` must elevate to enable this registry edit when prompting the user.
+
+> 📓 Technical Details:
+> A registry key is preferred over an environment variable because it is an admin-protected context. We don't want user-local controllable behavior to influence/break the behavior of an administrative installer, as that is a security concern. A policy key may also be used, but a policy key is essentially equivalent to a registry key in this context.
+>
+> On Unix, we could set this via a protected sentinel file.
+
+3. Under step 2, the `PATH` is updated to a dotnet application host (dotnet.exe) that contains the local installs managed by `bootstrapper`. Visual Studio and the other installers will no longer set the System `PATH` until the registry key or sentinel change is reverted. The option to undo this would be available, by removing the key, perhaps via an `uninstall all` or `reset` CLI feature from `bootstrapper`. The exact semantics of such are not pertinent here.
+
+4. The existing `global.json` `paths` feature would still enable other scenarios like for Visual Studio to specify any SDK. In this sense, it is already possible to enable local .NET SDKs to work with VS.
+
+5. The user is prompted to see if they would like to install the local versions of the administrative installs currently managed by Visual Studio/MU/Package Manager, as well as the update option they prefer for those installs (ala global.json `rollFoward` values, as described in the cli-experience document.)
+
+> 📓 Technical Details: Admin Install Detection
+> We can check the registry key hive as Visual Studio includes a reference count of which SDKs it manages and the file system admin locations in other scenarios. This prevents disrupting Visual Studio functionality as the local installs would match all of the SDKs that it had previously depended upon. This does consume extra disk space. If the user installs a newer SDK, Visual Studio would use that SDK, which is consistent with current behavior.
+
+> 📓 Technical Details: Update
+> Upon the action of step 5, we would need to take on the responsibility of updating these SDKs. Managing a daemon is resource-intensive, so it may not be optimal. SDL requirements specify a required update policy; we cannot hook into MU and IT administrators rely on administrative install behaviors. One approach is to mimic the update behavior of Visual Studio upon every interaction with `bootstrapper` by suggesting updates and asking the user to specify an update cadence when the new local mode is applied. IT administrators should be able to enforce a policy that requires a specific update cadence, and the provided update cadence should be compliant with SDL requirements, such as checking for updates on a daily cadence when the tool is used. Visual Studio would need to trigger this check as well to maintain compliance.
```

Comment:
To start with, I believe it's enough to notify the user of available updates and allow them to install those updates (via an explicit action).  I don't think we need any sort of automatic update in order to comply with the SDK (or to have a good experience).

I do think that the update notifications should not be limited to interactions with the bootstrapper or installation related commands if we make them part of `dotnet`.  Typical `dotnet` development commands such as `dotnet build` should notify the user that an update is available.

I also don't think we need (at least to start with) any IT policy around this, beyond possibly disabling user installs entirely.  If IT wants to manage .NET installs, then they should probably be using admin installs.